bin.js

```diff
@@ -387,3 +437,32 @@ function checkForUpdates () {
     defer: false
   })
 }
+
+function tarAndUploadFile (uploadURL, authToken, email) {
+  return async (filename) => {
+    // filename may either be .clinic-doctor.html or the data directory
+    // .clinic-doctor
+    const filePrefix = path.join(filename).replace(/\.html$/, '')
+
+    console.log(`Uploading private data for user ${email} for ${filePrefix} and ${filePrefix}.html to ${uploadURL}`)
+
+    return tarAndUploadPromisified(path.resolve(filePrefix), uploadURL, authToken)
+  }
+}
+
+function processAsk (args, authMethod) {
+  return authMethod(args['upload-url'])
+    .then(function (authToken) {
+      const { email } = jwt.decode(authToken)
+      const uploadURL = args['upload-url']
+      const uploaderFunc = tarAndUploadFile(uploadURL, authToken, email)
+
+      return new Promise((resolve, reject) => {
+        async.eachSeries(args._, uploaderFunc, err => err ? reject(err) : resolve())
```

Comment:
a better approach is to call `args._map()` and use `Promise.all()` on that one.

Reply:
refactored